sdk/database/newdbplugin/grpc_client.go

```diff
@@ -0,0 +1,249 @@
+package newdbplugin
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/hashicorp/vault/sdk/database/newdbplugin/proto"
+	"github.com/hashicorp/vault/sdk/helper/pluginutil"
+)
+
+var (
+	_ Database = gRPCClient{}
+
+	ErrPluginShutdown = errors.New("plugin shutdown")
+)
+
+type gRPCClient struct {
+	client  proto.DatabaseClient
+	doneCtx context.Context
+}
+
+func (c gRPCClient) Initialize(ctx context.Context, req InitializeRequest) (InitializeResponse, error) {
+	rpcReq, err := initReqToProto(req)
+	if err != nil {
+		return InitializeResponse{}, err
+	}
+
+	rpcResp, err := c.client.Initialize(ctx, rpcReq)
+	if err != nil {
+		return InitializeResponse{}, fmt.Errorf("unable to initialize: %s", err.Error())
+	}
+
+	return initRespFromProto(rpcResp)
+}
+
+func initReqToProto(req InitializeRequest) (*proto.InitializeRequest, error) {
+	config, err := json.Marshal(req.Config)
```

Comment:
It seems that json encoding/decoding might not work properly across messages. Consider using [`protojson`](https://godoc.org/google.golang.org/protobuf/encoding/protojson).

Reply:
This won't work here as a drop-in replacement. `protojson.Marshal` takes in a `proto.Message` type, not `interface{}`.

What is the problem for encoding/decoding here? I'm not clear what the issue is.